src/deployments/maintainers.py

```diff
@@ -0,0 +1,218 @@
+
+import argparse
+import datetime as dt
+import subprocess
+import sys
+from collections import Counter
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, Iterable, List, Sequence
+
+
+def _discover_repo_root() -> Path:
+    """Return the repository root by walking up from this module."""
+
+    module_path = Path(__file__).resolve()
+    for candidate in module_path.parents:
+        if (candidate / "helm").exists():
+            return candidate
+    return Path.cwd().resolve()
+
+
+REPO_ROOT = _discover_repo_root()
+CHART_ROOTS: Sequence[Path] = (
+    REPO_ROOT / "helm" / "applications",
+    REPO_ROOT / "helm" / "common",
+)
+LOG_FORMAT = "%H%x1f%an%x1f%ae%x1f%ad"
+MAX_MAINTAINERS = 2
+MONTHS_WINDOW = 12
+
+
+@dataclass
+class Maintainer:
+    """Single maintainer entry for Chart.yaml."""
+
+    name: str
+    email: str
+
+    def as_yaml(self) -> List[str]:
+        """Render maintainer dictionary in YAML list form."""
+        lines = [f"  - name: {self.name}"]
+        if self.email:
+            lines.append(f"    email: {self.email}")
+        return lines
+
+
+@dataclass
+class Commit:
+    """Git commit metadata used for maintainer inference."""
+
+    author: str
+    email: str
+    date: dt.datetime
+
+
+def read_chart_paths() -> List[Path]:
+    """Return all chart directories that contain a Chart.yaml file."""
+    charts: List[Path] = []
+    for root in CHART_ROOTS:
+        if not root.exists():
+            continue
+        for chart_file in sorted(root.rglob("Chart.yaml")):
+            charts.append(chart_file.parent)
+    return charts
+
+
+def git_log_for_path(path: Path) -> List[Commit]:
+    """Read commit history for a chart path."""
+    command = [
+        "git",
+        "log",
+        f"--pretty=format:{LOG_FORMAT}",
+        "--date=iso-strict",
+        "--",
+        str(path.relative_to(REPO_ROOT)),
+    ]
+    result = subprocess.run(
+        command,
+        cwd=REPO_ROOT,
+        check=False,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        text=True,
+    )
+    if result.returncode != 0:
+        raise RuntimeError(f"git log failed for {path}: {result.stderr.strip()}")
+
+    commits: List[Commit] = []
+    for line in result.stdout.splitlines():
+        if not line.strip():
+            continue
+        _, author, email, raw_date = line.split("\x1f")
+        try:
+            commit_date = dt.datetime.fromisoformat(raw_date.replace("Z", "+00:00"))
+        except ValueError:
+            continue
+        commits.append(Commit(author=author.strip(), email=email.strip(), date=commit_date))
+    return commits
+
+
+def select_recent_commits(commits: Sequence[Commit], months: int) -> List[Commit]:
+    """Filter commits to the trailing N months relative to the latest commit."""
+    if not commits:
+        return []
+    latest = max(commit.date for commit in commits)
+    cutoff = latest - dt.timedelta(days=months * 30)
```

Comment:
This is probably too loose.  I would even just say `months * 12`.

Reply:
Agreed, I will move the cutoff to 1 year.